main/sceneManager.cpp

```diff
@@ -1,26 +1,66 @@
 #include "main/sceneManager.h"
 
-using Tri = objects::TriangleObj;
+SceneManager::SceneManager()
+{
+    createCollectors();
+}
+
+std::vector<objects::EnergyCollector *> SceneManager::getEnergyCollectors()
+{
+    std::vector<objects::EnergyCollector *> temp;
+    temp.reserve(_energyCollectors.size());
+    for (const auto &t : _energyCollectors)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
+
+std::vector<objects::TriangleObj *> SceneManager::getModelTriangles()
+{
+    std::vector<objects::TriangleObj *> temp;
+    temp.reserve(_modelTriangles.size());
+    for (const auto &t : _modelTriangles)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
 
-void IamSimulator()
+// PRIVATE METHODS
+
+void SceneManager::createCollectors()
 {
-    SceneManager manager;
-    CHECK(manager.loadModel("123231")) << "Cannot load model";
+    _energyCollectors.reserve(constants::kCollectors);
 
-    std::vector<Tri> triangles = manager.getModelTriangles();
-    generators::PointSource source;
+    const double collectorRadius = static_cast<double>(constants::kSimulationRadius) / 2;
+    double alphaIncrement;
 
-    for (int i = 0; i < 1000000; i++)
+    if (constants::kCollectors % 2 == 0)
+    {
+        alphaIncrement = 2 * constants::kPi * 180 / (static_cast<double>(constants::kCollectors) / 2 * 360);
+    }
+    else
     {
-        auto ray = source.generateRay(1, 2);
+        alphaIncrement = 2 * constants::kPi * 180 / ((static_cast<double>(constants::kCollectors) - 1) / 2 * 360);
+    }
 
-        for (const Tri &triangle : triangles)
+    for (double alpha = 0; alpha <= constants::kPi; alpha += alphaIncrement)
+    {
+        double cur = collectorRadius * std::sin(alpha);
+        double z = collectorRadius * std::cos(alpha);
+
+        core::Vec3 xAxesOrigin(cur, 0, z);
+        core::Vec3 yAxesOrigin(0, cur, z);
+
+        if (xAxesOrigin == core::Vec3(0, 0, constants::kSimulationRadius / 2))
+        {
+            _energyCollectors.push_back(std::make_unique<objects::EnergyCollector>(objects::EnergyCollector(xAxesOrigin)));
+        }
+        else
         {
-            core::RayHitData hitData;
-            if (triangle.hitObject(ray, 123123, &hitData))
-            {
-                ///
-            }
+            _energyCollectors.push_back(std::make_unique<objects::EnergyCollector>(objects::EnergyCollector(xAxesOrigin)));
```

Comment:
No need to repead the type name so many times: what you have here does this:

1. Creates a new temp. object EnergyCollector with xAxesOrigin
2. Creates a new object on the heap, in make_unique
3. calls the copy constructor

make_unique already calls the constructor, so write it like this:

```
_energyCollectors.push_back(std::make_unique<objects::EnergyCollector>(xAxesOrigin));
```

Reply:
fixed